h5netcdf/core.py

```diff
@@ -95,16 +96,25 @@ def _lookup_dimensions(self):
             return (child_name,)
 
         dims = []
+        phony_dims = Counter()
```

Comment:
Nit (feel free to ignore): `Counter` is fine, but it is a little slow and does a lot of magic. I usually stick to the simpler `collections.defaultdict(int)`.

Reply:
Everything which positively affects performance is better. I'll try to get your suggestion working.